### PR TITLE
Fix: persist users classifications

### PR DESCRIPTION
**Pull request by JWGmeligMeyling**

### CODE REVIEW

File: src/main/java/me/moodcat/api/SongAPI.java [hunk 1]
```diff
@@ -119,16 +129,44 @@ public SongModel getSongById(@PathParam("id") final int id) {
...
+
+        if(classificationDAO.exists(user, song)) {
+            /*
+             * This should not happen, because after the initial classification
+             * the song will not be available for classification (and this endpoint
+             * should throw an IllegalArgumentException instead). However, this
+             * endpoint is currently also abused for the classification game.
+             *
+             * In order to prevent users to abuse this to gain points with fake
+             * classifications on the same song, we ignore duplicate classifications.
```
Comment (JeremybellEU) [inline src/main/java/me/moodcat/api/SongAPI.java]: An endpoint exists for the game itself (approachSong method below), so we might have to fix the issue there too. Or deduplicate these 2 endpoints in order to save ourselves some time.

**Comment by JeremybellEU** [comment]
Looks good, but we have to fix the import order changes though. Will fix them later with Eclipse.

File: src/main/java/me/moodcat/api/SongAPI.java (hunk 1 above)
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/SongAPI.java]: Yes, but this is a front end issue. When its fixed on the frontend, we can optimize the other endpoint as well.

Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/api/SongAPI.java]: (and the frontend issue is in no way an easy fix, it requires quite a large overhaul and we need this fixed fast)

**Comment by JWGmeligMeyling** [comment]
Formattted

**Comment by JeremybellEU** [comment]
:+1:

**Pull request merged**